homeassistant/components/sensor/bom.py

```diff
@@ -211,17 +213,41 @@ def get_reading(self, condition):
         for the latest value that is not `-`.
 
         Iterators are used in this method to avoid iterating needlessly
-        iterating through the entire BOM provided dataset.
+        through the entire BOM provided dataset.
         """
         condition_readings = (entry[condition] for entry in self._data)
         return next((x for x in condition_readings if x != '-'), None)
 
     @Throttle(MIN_TIME_BETWEEN_UPDATES)
     def update(self):
-        """Get the latest data from BOM."""
+        """Get the latest data from BOM.
+
+        Check if it is > 35 minutes since the last reading came in.
+        The readings are timestamped on the hour/half hour and the
+        data generally refreshes 35 minutes after that.
+        You don't want to just throttle for 35 mins as that doesn't take
+        into account how far you currently are from the last reading time.
+        """
+        if self._lastupdate != 0 and \
```

Comment:
Perhaps you could extract this boolean logic into a helper method on the class, might make it easier to read (and you could test it separately too!)

```python
def should_update(self):
   now = datetime.datetime.now()
   return self._last_updated is None or self._last_updated + datetime.timedelta(minutes=35) < now

def update(self):
   if not self.should_update():
      _LOGGER.debug(...)
      return
   
   ...
   ... do update things
```